src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java

```diff
@@ -191,6 +192,55 @@ public void requestHandlerTest() throws IOException {
 		assertEquals("Listener failed to accept connections exactly once from sender in http mode.", 1, handlerExecuted.get());
 	}
 	
+	@Test
+	public void parseRequestPathAndQueryTest() throws IOException, URISyntaxException {
+		CompletableFuture<Void> requestReceived = new CompletableFuture<Void>();
+		String extraPath = "/extraPath/extraPath2";
+		String extraQuery = "queryKey=queryValue&queryKey2=queryValue;"; // mixing in a special char just for fun
+		
+		listener.setRequestHandler((context) -> {
+			try {
+				assertNotNull("Listener should have received a valid http context.", context);
+				assertNotNull("Listener should have received a valid http request.", context.getRequest());
+				
+				URI requestUri = context.getRequest().getUri();
+				assertNotNull("Listener should have a URI from request", requestUri);
+				assertEquals(
+						"The path wasn't the expected value", 
+						"/bailiu-relay-hc/bailiu-relay-hc" + extraPath, 
```

Comment:
>bailiu-relay-hc/bailiu-relay-hc [](start = 8, length = 31)

This won't work for other people, right?

Reply:
I think it should be based on TestUtil.ENTITY_PATH

---
In reply to: [263873685](https://github.com/Azure/azure-relay-java/pull/31#discussion_r263873685) [](ancestors = 263873685)

Reply:
Yes it should be TestUtil.ENTITY_PATH, although I find it weird why it's appended again.

---
In reply to: [263873685](https://github.com/Azure/azure-relay-java/pull/31#discussion_r263873685) [](ancestors = 263873685)

Reply:
Is that a bug?

---
In reply to: [263976580](https://github.com/Azure/azure-relay-java/pull/31#discussion_r263976580) [](ancestors = 263976580,263873685)